torch/_inductor/fx_passes/group_batch_fusion.py

```diff
@@ -156,6 +156,78 @@ def fuse(self, graph, subset):
             graph.erase_node(original_mm)
 
 
+class BatchLayernormFusion(BatchFusion):
+    """
+    Batch layer norm fusion in pre grad pass
+    """
+
+    def match(self, node):
+        if CallFunctionVarArgs(torch.nn.functional.layer_norm).match(node):
+            input_node = get_arg_value(node, 0, "input")
+            group_key = (
+                "batch_layernorm",
+                str(input_node.meta["example_value"].shape)
+                if "example_value" in input_node.meta
+                else "",
+                str(get_arg_value(node, 1, "normalized_shape")),
+                str(get_arg_value(node, 4, "eps")),
+            )
+        else:
+            group_key = None
+        return group_key
+
+    def fuse(self, graph, subset):
+        group_inputs = []
+        group_shapes = []
+        group_weights = []
+        group_biases = []
+        group_epss = []
+        group_nodes = []
+        for node in subset:
+            group_nodes.append(node)
+            group_inputs.append(get_arg_value(node, 0, "input"))
+            group_shapes.append(get_arg_value(node, 1, "normalized_shape"))
+            group_weights.append(get_arg_value(node, 2, "weight"))
```

Comment:
`weight` is an optional arg.   What happens if some have weight set and others don't?